build/typescript-options.js

```diff
@@ -0,0 +1,40 @@
+exports.base = function() {
+  return {
+    "target": "es5",
+    "module": "es2015",
+    "experimentalDecorators": true,
+    "emitDecoratorMetadata": false,
+    "moduleResolution": "node",
+    "stripInternal": true,
+    "preserveConstEnums": true,
+    "declaration": true,
+    "noExternalResolve": false,
+    "removeComments": true,
+    "lib": ["es2015", "dom"],
+    "typescript": require('typescript')
+  }
+}
+
+exports.commonjs = function() {
+  var options = exports.base();
+  options.module = 'commonjs';
+  return options;
+};
+
+exports.amd = function() {
+  var options = exports.base();
+  options.module = 'amd';
+  return options;
+};
+
+exports.system = function() {
+  var options = exports.base();
+  options.module = 'system';
+  return options;
+};
```

Comment:
I'd remove separate functions for different module formats (`commonjs`, `amd` and `system`) and instead add one function, that returns compiler options for given module format - smth like this:
```javascript
exports.compilerTsOptionsForModule(module) {
  var options = exports.base();
  options.module = module;
  return options;
}
```
Then the build.js also becomes much shorter and easier to understand :)